test/e2e/suite/conformance/certificates/tests.go

```diff
@@ -605,5 +768,119 @@ func (s *Suite) Define() {
+			testCertificate := &cmapi.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testcert",
+					Namespace: f.Namespace.Name,
+				},
+				Spec: cmapi.CertificateSpec{
+					SecretName: "testcert-tls",
+					DNSNames:   []string{"google.com"},
+					IssuerRef:  issuerRef,
+				},
+			}
+
+			By("Creating a Certificate")
+			err := f.CRClient.Create(ctx, testCertificate)
+			Expect(err).NotTo(HaveOccurred())
+
+			notReadyCondition := cmapi.CertificateCondition{
+				Type:   cmapi.CertificateConditionReady,
+				Status: cmmeta.ConditionFalse,
+			}
+			Eventually(testCertificate, "30s", "1s").Should(HaveCondition(f, notReadyCondition))
+			Consistently(testCertificate, "1m", "10s").Should(HaveCondition(f, notReadyCondition))
+		}, featureset.OnlySAN, featureset.CheckValidACMEDNSName)
+
+		s.it(f, "should allow updating an existing certificate by adding a new dns name", func(issuerRef cmmeta.ObjectReference) {
+			testCertificate := &cmapi.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testcert",
+					Namespace: f.Namespace.Name,
+				},
+				Spec: cmapi.CertificateSpec{
+					SecretName: "testcert-tls",
+					DNSNames:   []string{e2eutil.RandomSubdomain(s.DomainSuffix)},
+					IssuerRef:  issuerRef,
+				},
+			}
+			validations := validation.CertificateSetForUnsupportedFeatureSet(s.UnsupportedFeatures)
+
+			By("Creating a Certificate")
+			err := f.CRClient.Create(ctx, testCertificate)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for the Certificate to be ready")
+			_, err = f.Helper().WaitForCertificateReady(f.Namespace.Name, "testcert", time.Minute*5)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Sanity-check the issued Certificate")
+			err = f.Helper().ValidateCertificate(f.Namespace.Name, "testcert", validations...)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Getting the latest version of the Certificate")
+			cert, err := f.Helper().CMClient.CertmanagerV1().Certificates(f.Namespace.Name).Get(context.TODO(), "testcert", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Adding an additional dnsName to the Certificate")
+			newDNSName := e2eutil.RandomSubdomain(s.DomainSuffix)
+			cert.Spec.DNSNames = append(cert.Spec.DNSNames, newDNSName)
+
+			By("Updating the Certificate in the apiserver")
+			err = f.CRClient.Update(context.TODO(), cert)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for the Certificate Ready condition to be updated")
+			_, err = f.Helper().WaitForCertificateReadyUpdate(cert, time.Minute*5)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Sanity-check the issued Certificate")
+			err = f.Helper().ValidateCertificate(f.Namespace.Name, "testcert", validations...)
+			Expect(err).NotTo(HaveOccurred())
+		}, featureset.OnlySAN)
+
+		s.it(f, "should allow updating an existing failing certificate that had an incorrect dns name", func(issuerRef cmmeta.ObjectReference) {
```

Comment:
What is incorrect about this DNS name? Again, I think this is specific to ACME and should be moved to the issuer specific tests.

Reply:
I'll move this back to the issuer specific test suite.
I'll also look into adding a more general test here that checks for a wrong DNS Name (eg. wrongly formatted) if it can still recover.